### DISCUSSION TITLE
We're trying out Github Discussions... thoughts???

### DISCUSSION THREAD

**Discussion by rhukster** [getgrav/grav#3097] [Ideas]
We have been using [Discourse](https://discourse.getgrav.org/) for a while now and while it’s a solid forum platform, I find that I personally don’t visit it enough as my attention is already spread between Github for issues and Discord for chatting. Github has recently made its own discussions platform publicly available, and we enabled this on the getgrav.org repo to take a look at it. It’s no way near as feature-rich as Discourse, but the major benefit is the tight integration into Github for tickets, pull requests, etc. That simplicity also makes it much easier to manage and maintain going forward, It just seems like a better fit.

What do you, the community, think about centralizing discussions directly on the `getgrav` Github project???

**Comment by hughbris** [comment]
I think tight integration is the killer feature. I also tend to forget about discourse because it's another site.

I don't know if this is a pro or con, we might get less frutstratingly naive newbie questions from casual, code-averse types who should perhaps be directed to Wix or Wordpress, or be encouraged to check docs first.

Centralising is essential. How do you plan to handle years of useful discourse content? I know you imported Muut into Discourse with mixed success.

**Reply by hughbris** [reply]
Supplementary: do you think questions about plugins should be in discussion under their separate repos? Discord gets a lot of those.

**Reply by w00fz** [reply]
> How do you plan to handle years of useful discourse content? I know you imported Muut into Discourse with mixed success.

I don't think it's realistically feasible or convenient for us to import from Discourse into GitHub. My idea was to put Discourse into read-only mode, keep it for the vast amount of useful information available, but then moving forward use GitHub discussions..

I think a Plugins category would be useful to have and frankly we could add categories as needed to keep it all tidy and organized.

**Reply by hughbris** [reply]
I'm assuming there's a script somewhere for the import. A read-only archive is probably better on aggregate on reflection.

Not suggesting so much a plugins category here (although that could be a catchall), more setting up discussions on each plugin repo and encouraging posters to move relevant threads over there so they sit with the code, issues etc. It's a better resting place, but arguably more complex to keep in line. Order tends to be the less natural option :(

**Reply by w00fz** [reply]
We can test how the Discussions here go, I see this being the centralized place anyway. We just got the Discussions feature unlocked, it is fresh out of GitHub over.
If this is well received we can definitely apply it to other repos.

**Reply by rhukster** [reply]
There is no import capability at all. We would definitely look at it if there was. Already commented on it: https://github.com/github/feedback/discussions/36

**Reply by jeremycherfas** [reply]
> encouraged to check docs first.

I would thoroughly support this if the docs were a little more approachable by us pesky "casual, code-averse types". We may not even be code-averse, more specific code-ignorant.

It is a perennial problem I know, which is by no means exclusive to Grav, that the docs are maintained by people who know the thing inside out and have a justifiably hard time putting themselves in the shoes of someone less experienced or knowledgeable than they are.

Chattier apps are great for quick back and forth when you find someone who is willing and able to help.

**Reply by hughbris** [reply]
> I would thoroughly support this if the docs were a little more approachable by us pesky "casual, code-averse types". We may not even be code-averse, more specific code-ignorant.

I absolutely take your point about the docs. To explain my accusation, which actually wasn't, I hold the opinion that if you freak out when you see code, you are going to have a bad time building a Grav site. I'm not gatekeeping, that's how it is right now. There are friendlier platforms. Sometimes I feel that _"do you know how a website works?"_ would be an appropriate question to some posters.

It's rare that posters say _"I checked the docs and couldn't find it / couldn't understand them"_. That tells me they only want to be spoonfed. Chat and forums each have their strengths, I agree.

**Reply by jeremycherfas** [reply]
And I'll take your point, and make a snippet of that "I checked the docs ..." for future use.

**Comment by jeremycherfas** [comment]
Is there any way to get a feed from the top-level discussion page, so one could see what was new without having to visit?

**Reply by rhukster** [reply]
From preliminary research it appears there is an api method to get a list of recent discussions.  So it means that it’s possible to turn this into rss to get a feed.

**Reply by jeremycherfas** [reply]
Thanks. I will investigate further.

**Comment by paulhibbitts** [comment]
Thanks for trying this out @rhukster @w00fz , from what I see I like this approach as it brings things closer to GitHub. As well, GitHub in general seems to have a lot of forward-velocity right now, so I would tend to see this new Discussion feature get more and more love resources wise🙂

**Reply by rhukster** [reply]
I think that it will be used a great deal more as many in the community already spend a lot of time in GitHub, so it's easy to check.  Also because it's so basic, it's actually more appealing to me.  Not something else we have to learn, manage, and keep running. GitHub API integration is actually a huge benefit too.  It means there's a good chance we can integrate it with Github Actions workflows in the future.

**Reply by paulhibbitts** [reply]
I also just noticed that my GitHub Notifications showed a new item when you replied to this message, nice!

PS: Also got an email notification. Trying out the Mac App https://www.gitify.io/ as well.

**Comment by w00fz** [comment]
Btw https://github.com/getgrav/grav/discussions/3100 - This is a Question / Answer discussion format. It gets enabled under certain categories. Right now it's Help and Development. My understanding is this will work similarly to stackoverflow, if you guys want to test this out as well, comment in it.

**Reply by paulhibbitts** [reply]
Sure, here is my comment/answer 🙂

**Comment by Sogl** [comment]
Nice feature, i don't like Discourse forum design etc and I get to it only through a search engine.

**Comment by gizmecano** [comment]
I do not yet have a clear opinion on the question, but IMHO there are two potential obstacles:

1. lose previous discussions, reflections, questions and solutions currently stored on Discourse: in this case, maybe it would be useful to look for a solution to migrate the content from Discourse to GitHub as it had been done with Muut
2. neglect some of the users registered on the Discourse forum and who do not wish to register on a new network like GitHub  (which in many aspects appears to be more complex to use than Discourse) just to follow simple discussions on general questions

Nevertheless, it is clear that such a direct integration of a discussion system within the development repository itself can certainly bring huge benefits.
